src/new-components/modal/modal.stories.mdx

```diff
@@ -77,10 +82,30 @@ behaviour, see [ARIA: dialog role](https://developer.mozilla.org/en-US/docs/Web/
         name="Modal with standard actions footer"
     >
         <ModalStoryStateProvider>
-            <Button variant="primary">Open modal</Button>
+            <Button variant="primary" action="open">
+                Open modal
+            </Button>
             <Modal aria-label="Modal with standard actions footer">
                 <ModalHeader>
-                    <Heading level="1">Modal with standard actions footer</Heading>
+                    <Columns>
```

Comment:
I think this fixes the misaligned modal header:

```suggestion
                    <Columns alignY="center">
```

Although, IMO, this extra button should be put in the header button slot, instead of putting it in the title slot. Something along these lines:

```jsx
<ModalHeader
  button={
    <Columns space="small">
      <Column><CustomButton /></Column>
      <Column><ModalCloseButton /></Column>
    </Columns>
  }
>
  <Heading level="1">Modal with standard actions footer</Heading>
</ModalHeader>
```